packages/permissionless/accounts/etherspot/signerToEtherspotSmartAccount.ts

```diff
@@ -0,0 +1,479 @@
+                TPrimaryType,
+                TChain,
+                undefined
+            >(client, {
+                account: viemSigner,
+                ...typedData,
+                accountAddress,
+                chainId
+            })
+
+            return concatHex([
+                getEcdsaRootIdentifier(ecdsaValidatorAddress),
+                signature
+            ])
+        },
+        client: client,
+        publicKey: accountAddress,
+        entryPoint: entryPointAddress,
+        source: "etherspotSmartAccount",
+
+        // Get the nonce of the smart account
+        async getNonce() {
+            const key = getNonceKeyWithEncoding(
+                ecdsaValidatorAddress
+                // @dev specify the custom nonceKey here when integrating the said feature
+                /*, nonceKey */
+            )
+            return getAccountNonce(client, {
+                sender: accountAddress,
+                entryPoint: entryPointAddress,
+                key
+            })
+        },
+
+        // Sign a user operation
+        async signUserOperation(userOperation) {
+            const hash = getUserOperationHash({
+                userOperation: {
+                    ...userOperation,
+                    signature: "0x"
+                },
+                entryPoint: entryPointAddress,
+                chainId: chainId
+            })
+            const signature = await _signMessage(client, {
+                account: viemSigner,
+                message: { raw: hash }
+            })
+
+            return signature
+        },
+
+        // Encode the init code
+        async getInitCode() {
+            if (smartAccountDeployed) return "0x"
+
+            smartAccountDeployed = await isSmartAccountDeployed(
+                client,
+                accountAddress
+            )
+
+            if (smartAccountDeployed) return "0x"
+
+            const _factoryAddress = metaFactoryAddress
+            return concatHex([_factoryAddress, await generateInitCode()])
+        },
+
+        async getFactory() {
+            if (smartAccountDeployed) return undefined
+
+            smartAccountDeployed = await isSmartAccountDeployed(
+                client,
+                accountAddress
+            )
+
+            if (smartAccountDeployed) return undefined
+
+            return metaFactoryAddress
+        },
+
+        async getFactoryData() {
+            if (smartAccountDeployed) return undefined
+
+            smartAccountDeployed = await isSmartAccountDeployed(
+                client,
+                accountAddress
+            )
+
+            if (smartAccountDeployed) return undefined
+
+            return generateInitCode()
+        },
+
+        // Encode the deploy call data
+        async encodeDeployCallData(_) {
+            throw new Error("Simple account doesn't support account deployment")
+        },
+
+        // Encode a call
+        async encodeCallData(_tx) {
```

Comment:
Can you change `_tx -> tx`. We use `_` to denote unused variables.

Reply:
Changes done.